server/api/api/endpoints/artifacts.py

```diff
@@ -74,7 +74,7 @@ async def store_artifact(
         "Storing artifact", project=project, uid=uid, key=key, tag=tag, iter=iter
     )
     await run_in_threadpool(
-        server.api.crud.Artifacts().store_artifact,
+        server.api.crud.Artifacts(auth_info=auth_info).store_artifact,
```

Comment:
u can't pass auth for constructing the artifacts object when it is a singleton or else you will leak first inbound request authentication details for any consecutive requests.